### PR TITLE
Docs fixes and add to README

### PR DESCRIPTION
**Pull request by BSnelling**

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -2,7 +2,103 @@
...
+The objective of the model is to solve a simple energy balance problem, where a set of loads (demands) need to be met by generation (supply), for the minimum possible cost.
+Assume we have an instance of a system (see [`SystemRT`](@ref) for details of all the components in a `SystemRT`).
+
+```julia
+using AxisKeys
+using FullNetworkSystems
+using HiGHS
+using JuMP
+
+system = SystemRT(components...)
```
Comment (mcmcgrath13) [inline README.md]: what is `components...` here?

File: README.md [hunk 2]
```diff
@@ -2,7 +2,103 @@
...
+units = keys(get_generators(system))
+datetimes = get_datetimes(system)
+
+@variable(model, generation[u in units, t in datetimes] >= 0)
+
+load = get_loads(system)
+@constraint(
+    model,
+    energy_balance[t in datetimes],
+    sum(model[:generation][u, t] for u in units) == sum(load(l, t) for l in axiskeys(load, 1))
```
Comment (mcmcgrath13) [inline README.md]: where does `load` come from? or is this jump magic introducing it?

File: README.md [hunk 3]
```diff
@@ -2,7 +2,103 @@
...
+    add_to_expression!(cost, prices(u, t), model[:generation][u, t])
+end
+
+@objective(model, Min, cost)
+
+set_optimizer(model, HiGHS.Optimizer)
+optimize!(model)
+```
+
+## Additional Features of the System
```
Comment (mcmcgrath13) [inline README.md]: I'd maybe move this section first and drop the `Additional`

File: README.md [hunk 4]
```diff
@@ -2,7 +2,103 @@
...
+
+This package defines a set of types and data structures to represent an energy grid and the components within it.
+The types defined can be used to build Optimal Power Flow and associated optimisation problems e.g. unit commitment and economic dispatch.
+The data structures are designed so that the relevant data can be easily accessed and used to formulate these optimisation problems.
+
+## Example Modelling Problem
+
+The following is an example of how the data stored in a `SystemRT` object can be used to build a JuMP model.
+The objective of the model is to solve a simple energy balance problem, where a set of loads (demands) need to be met by generation (supply), for the minimum possible cost.
+Assume we have an instance of a system (see [`SystemRT`](@ref) for details of all the components in a `SystemRT`).
```
Comment (mcmcgrath13) [inline README.md]: the `@ref` links don't work in a readme - replace these with stable docs links

**Review by mcmcgrath13 — COMMENTED** [review]
Lots more context 🎉 📖

File: README.md (hunk 3 above)
Comment (mcmcgrath13) [reply README.md]: or maybe `Main` or `Primary`

File: README.md (hunk 1 above)
Comment (BSnelling) [reply README.md]: This should have a bit more context now.  The example model appears after a section describing all the components and here the names of the components are listed (although I still haven't included all the code needed to construct them).

File: README.md (hunk 2 above)
Comment (BSnelling) [reply README.md]: `load` is defined above from `get_load(system)`.  It is a `KeyedArray` so uses `()` for indexing by the names of the keys within the `@constraint` definition.

File: README.md (hunk 3 above)
Comment (BSnelling) [reply README.md]: Done

File: README.md (hunk 4 above)
Comment (BSnelling) [reply README.md]: Links are to the `dev` docs for now, until I manage to fix the stable docs deployment.

File: README.md [hunk 5]
```diff
@@ -2,7 +2,140 @@
...
+The objective of the model is to solve a simple energy balance problem, where a set of loads (demands) need to be met by generation (supply), for the minimum possible cost.
+Assume we have built all the components of the system described in the previous sections so that we can construct an instance of a system (see [`SystemRT`](https://invenia.github.io/FullNetworkSystems.jl/stable/#FullNetworkSystems.SystemRT) for specific details of the components in a `SystemRT`).
+
+```julia
+using AxisKeys
+using FullNetworkSystems
+using HiGHS
+using JuMP
+
+system = SystemRT(
```
Comment (mcmcgrath13) [inline README.md:104]: maybe just add a comment here on the simplifying assumption of this example

Comment (BSnelling) [reply README.md:104]: Added a comment, does it clarify things?

File: README.md (hunk 4 above)
Comment (mcmcgrath13) [reply README.md]: Looks good to me!

**Pull request merged**